contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -104,11 +105,22 @@ func (k *k8smScheduler) killTask(taskId string) error {
 }
 
 func (k *k8smScheduler) launchTask(task *podtask.T) error {
+	// create or update node first with current labels
+	s, found := k.internal.slaves.getSlave(task.Offer.Details().GetSlaveId().GetValue())
+	if !found || s == nil {
+		return fmt.Errorf("cannot find slave to launch tasks on: %s", task.Spec.AssignedSlave)
+	}
+	labels := node.SlaveAttributesToLabels(task.Offer.Details().Attributes)
+	err := s.createOrUpdateNode(k.internal.client, labels)
```

Comment:
is it possible that any other goroutine could be modifying this Slave record at the same time? launchTask() is already guarded by its callers. I'm more concerned with other users of Slave objects